diff --git a/setup.py b/setup.py
index 1973225..7e1a7f1 100644
--- a/setup.py
+++ b/setup.py
@@ -322,10 +322,7 @@ if __name__ == '__main__':
         install_requires=[
             'pynvml',
         ],
-        ext_modules=[
-            get_extension_deep_ep_cpp(),
-            get_extension_hybrid_ep_cpp()
-        ],
+        ext_modules=[get_extension_hybrid_ep_cpp()],
         cmdclass={
             'build_ext': BuildExtension
         },
